runtime/drivers/duckdb/model_executor_self.go

```diff
@@ -69,21 +111,37 @@ func (e *selfToSelfExecutor) Execute(ctx context.Context, opts *drivers.ModelExe
 		_ = olap.DropTable(ctx, stagingTableName)
 
 		// Create the table
-		createTableOpts := &drivers.CreateTableOptions{
-			View:         asView,
-			BeforeCreate: inputProps.PreExec,
-			AfterCreate:  inputProps.PostExec,
-		}
-		res, err := olap.CreateTableAsSelect(ctx, stagingTableName, inputProps.SQL, createTableOpts)
-		if err != nil {
-			_ = olap.DropTable(ctx, stagingTableName)
-			return nil, fmt.Errorf("failed to create model: %w", err)
+		if inputProps.DB != "" {
+			// special handling for ingesting from an external database
+			// not handling incremental use cases since ingesting from an external database is mostly for small,experimental use cases
```

Comment:
This is fine, but then perhaps add a check that returns an error if `.DB` is set when `opts.IncrementalRun` is true